src/FffGcodeWriter.cpp

```diff
@@ -1613,58 +1617,76 @@ bool FffGcodeWriter::processSingleLayerInfill(const SliceDataStorage& storage, L
         //Given that the original test worked for approximately 2x2cm models, this scaling by circumference should make it work for any size.
-        const double minimum_small_area = 0.4 * 0.4 * circumference / 40000;
+        const double minimum_small_area = 0.4 * 0.4 * static_cast<double>(circumference) / 40000;
         
         // This is only for density infill, because after generating the infill might appear unnecessary infill on walls
         // especially on vertical surfaces
         in_outline.removeSmallAreas(minimum_small_area);
-        
-        Infill infill_comp(pattern, zig_zaggify_infill, connect_polygons, in_outline, /*outline_offset =*/ 0
-            , infill_line_width, infill_line_distance_here, infill_overlap, infill_multiplier, infill_angle, gcode_layer.z, infill_shift, wall_line_count, infill_origin
-            , /*Polygons* perimeter_gaps =*/ nullptr
-            , /*bool connected_zigzags =*/ false
-            , /*bool use_endpieces =*/ false
-            , /*bool skip_some_zags =*/ false
-            , /*int zag_skip_count =*/ 0
-            , mesh.settings.get<coord_t>("cross_infill_pocket_size"));
-        infill_comp.generate(infill_polygons, infill_lines, mesh.cross_fill_provider, &mesh);
-    }
-    if (infill_lines.size() > 0 || infill_polygons.size() > 0)
+
+        constexpr size_t outline_offset = 0;
+        constexpr Polygons* perimeter_gaps = nullptr;
+        constexpr bool connected_zigzags = false;
+        constexpr bool use_endpieces = false;
+        constexpr bool skip_some_zags = false;
+        constexpr size_t zag_skip_count = 0;
+        wall_tool_paths.emplace_back(ToolPaths());
+
+        Infill infill_comp(pattern, zig_zaggify_infill, connect_polygons, in_outline, outline_offset, infill_line_width,
+                           infill_line_distance_here, infill_overlap, infill_multiplier, infill_angle, gcode_layer.z,
+                           infill_shift, wall_line_count, infill_origin, perimeter_gaps, connected_zigzags,
+                           use_endpieces, skip_some_zags, zag_skip_count,
+                           mesh.settings.get<coord_t>("cross_infill_pocket_size"));
+        infill_comp.generate(wall_tool_paths.back(), infill_lines, mesh.cross_fill_provider, &mesh);
+    }
+
+    const bool walls_generated = std::any_of(wall_tool_paths.cbegin(), wall_tool_paths.cend(), [](ToolPaths tp){ return !tp.empty(); });
+    if (!infill_lines.empty() || walls_generated)
     {
         added_something = true;
         setExtruder_addPrime(storage, gcode_layer, extruder_nr);
         gcode_layer.setIsInside(true); // going to print stuff inside print object
         std::optional<Point> near_start_location;
         if (mesh.settings.get<bool>("infill_randomize_start_location"))
         {
-            srand(gcode_layer.getLayerNr());
+            srand(static_cast<unsigned int>(gcode_layer.getLayerNr()));
```

Comment:
I believe so as well, will check once I'm behind a computer again

Reply:
I wanted to ask your opinion about the use of rand(). This function has limited randomness it a better option would be to use the <random> library and functions instead.

See https://wiki.sei.cmu.edu/confluence/plugins/servlet/mobile?contentId=88046554#content/view/88046554

Reply:
We should make one change at a time. If there's a bug with the uniform distribution it'll be hard to trace down. And we want to get things finished too.
Furthermore there are different levels of security for random number generators. The uniform distribution is not yet considered cryptographically secure, but does remove some modulo bias. But that also means it's more expensive to calculate. We don't need any security here since there is no attack vector. The worst that an attacker could do is cause the Z seams to align in a certain way.